cardano-api/src/Cardano/Api/IPC.hs

```diff
@@ -586,17 +592,26 @@ queryNodeLocalState :: forall mode result.
                     -> Maybe ChainPoint
                     -> QueryInMode mode result
                     -> IO (Either AcquiringFailure result)
-queryNodeLocalState connctInfo mpoint query = do
-    resultVar <- newEmptyTMVarIO
-    connectToLocalNode
-      connctInfo
+queryNodeLocalState connctInfo mpoint query =
+  runOopsInEither $ queryNodeLocalState_ connctInfo mpoint query
+
+queryNodeLocalState_ :: forall e mode result. ()
+  => e `CouldBe` AcquiringFailure
+  => LocalNodeConnectInfo mode
+  -> Maybe ChainPoint
+  -> QueryInMode mode result
+  -> ExceptT (Variant e) IO result
```

Comment:
The type-signature of an `oops`-enabled function is very similar to the one that is not.  There are only two differences:

* The return value has `ExceptT (Variant e)` instead of `ExceptT AcquiringFailure`.
* The function has an additional constraint `e ``CouldBe`` AcquiringFailure`

The `Variant e` means the function could return `0` or more errors.  To find out what errors the function can return, look at the `CouldBe` constraints.  Each one of those constraint is an error the function can raise.

The `oops` style code has some advantages over the non-oops style code.  We no longer have to worry about the ordering of errors compared to code like `ExceptT x (ExceptT y ...` or `Either x (Either y ...)`.  We don't have worry that it isn't compatible with `ExceptT y (Except x ..` or `ExceptT y`.

This advantage does not play out here yet.  Only in a future commit.  Other than that, there is very little novel about this type signature.  It uses type application which Haskell developers should be familiar with and class constraints.  The only novelty is we are using here is the use of constraints to specified what errors are thrown.

It's important to note that specifying errors in constraints is **useful**.  That's because we want to not worry about the order of errors and constraints is exactly the Haskell feature that offers this.  The use of constraints here is not an example of gratuitous abstraction, it is exactly the abstraction we need.